finetune/util/summaries.py

```diff
@@ -0,0 +1,13 @@
+import tensorflow as tf
+import re
+
+def make_summary_optimizer(optimizer, regex):
+    class SummaryOptimizer(optimizer):
+        def apply_gradients(self, grads_and_vars, *args, **kwargs):
```

Comment:
ah such a simple option, don't know why I hadn't thought of it. I think I was complicating things because I was trying to get norms of variable subsets, e.g. compare norms for all the positional variables aggregated vs. all the non-positional variables aggregated. but that's pretty trivial to build upon

Reply:
though for the sake of this weekend's experiments, I think keeping the summaries consistent across runs is probbaly more useful

Reply:
the pos variables vs. non-pos variables distinction is only useful for the sidekick model after all

Reply:
I've set up mlm_pretraining to do both for the sidekick model and just this one for the others